Packs/IvantiHeat/Integrations/IvantiHeat/IvantiHeat.py

```diff
@@ -0,0 +1,285 @@
+import dateparser
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' CONSTANTS '''
+TOKEN = demisto.params().get('token')
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+OBJECT_HEADERS = ['RecId', 'Subject', 'Name', 'Status', 'CreatedDateTime', 'Symptom', 'Urgency', 'OwnerTeam',
+                  'IncidentNumber', 'CreatedBy', 'Details', 'Owner', 'Category', 'Description', 'Priority',
+                  'Email', 'TypeOfIncident', 'ClosedDateTime', 'ActualCategory', 'SocialTextHeader']
+
+
+class Client(BaseClient):
+    """
+    Client will implement the service API, and should not contain any Demisto logic.
+    Should only do requests and return data.
+    """
+    def do_request(self, method, url_suffix, params=None, data=None, json_data=None, resp_type='json', files=None):
+
+        res = self._http_request(method, url_suffix, params=params, data=data, json_data=json_data, files=files,
+                                 resp_type=resp_type, ok_codes=(200, 201, 204), return_empty_response=True)
+
+        if resp_type == 'other' or isinstance(res, dict) or isinstance(res, list):
+            return res
+        return {}
+
+
+def test_module(client):
+    client.do_request('GET', 'odata/businessobject/incidents')
+    return 'ok'
+
+
+def generate_filter_param(rec_id, from_date, to_date):
+    if rec_id:
+        return f'recId eq \'{rec_id}\''
+
+    if not from_date and not to_date:
+        return ''
+    if from_date and not to_date:
+        return f'CreatedDateTime gt {from_date}'
+    if to_date and not from_date:
+        return f'CreatedDateTime lt {to_date}'
+    return f'CreatedDateTime gt {from_date} and CreatedDateTime lt {to_date}'
+
+
+@logger
+def objects_list_command(client, args):
+    object_type = args.get('object-type')
+    rec_id = args.get('rec-id')
+    from_date = args.get('from')
+    to_date = args.get('to')
+    limit = args.get('limit')
+    offset = args.get('offset')
+    query = args.get('search-query')
+
+    params = {'$orderby': 'CreatedDateTime desc'}
+    filter_txt = generate_filter_param(rec_id, from_date, to_date)
+
+    if filter_txt:
+        params['$filter'] = filter_txt
+    if limit:
+        params['$top'] = limit
+    if offset:
+        params['$skip'] = offset
+    if query:
+        params['$search'] = query
+
+    raw_res = client.do_request('GET', f'odata/businessobject/{object_type}', params=params)
+    if raw_res.get('value'):
+        human_readable = tableToMarkdown(f'{object_type} results', t=raw_res.get('value'), headers=OBJECT_HEADERS,
+                                         removeNull=True)
+
+        entry_context = {f'IvantiHeat.{object_type}(val.RecId===obj.RecId)': raw_res.get('value')}
+        return human_readable, entry_context, raw_res
+
+    return 'No records found', {}, raw_res
+
+
+@logger
+def update_object_command(client, args):
+    object_type = args.get('object-type')
+    rec_id = args.get('rec-id')
+    fields = args.get('fields')
```

Comment:
why not use JSON object same as the object for the create command (and use the same script to create the object) script

Reply:
I changed it, ivanti-heat-object-update is now getting JSON object